src/Compilers/CSharp/Test/Symbol/Symbols/ModuleInitializers/ModuleInitializersTests.cs

```diff
@@ -521,5 +657,119 @@ void validator(ModuleSymbol module)
                 Assert.NotNull(moduleType.GetMember<MethodSymbol>(".cctor"));
             }
         }
+
+        [Fact]
+        public void ModuleInitializerAttributeIncludedByConditionalAttribute()
+        {
+            string source = @"
+#define INCLUDE
+
+using System;
+using System.Runtime.CompilerServices;
+
+class C
+{
+    [ModuleInitializer]
+    internal static void M() => Console.WriteLine(""C.M"");
+}
+
+class Program 
+{
+    static void Main() => Console.WriteLine(""Program.Main"");
+}
+
+namespace System.Runtime.CompilerServices
+{
+    [System.Diagnostics.Conditional(""INCLUDE"")]
+    class ModuleInitializerAttribute : System.Attribute { }
+}
+";
+            CompileAndVerify(source, parseOptions: s_parseOptions, expectedOutput: @"
+C.M
+Program.Main");
+        }
+
+        [Fact]
+        public void ModuleInitializerAttributeExcludedByConditionalAttribute()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class C
+{
+    [ModuleInitializer]
+    internal static void M() => Console.WriteLine(""C.M"");
+}
+
+class Program 
+{
+    static void Main() => Console.WriteLine(""Program.Main"");
+}
+
+namespace System.Runtime.CompilerServices
+{
+    [System.Diagnostics.Conditional(""EXCLUDE"")]
+    class ModuleInitializerAttribute : System.Attribute { }
+}
+";
+            CompileAndVerify(source, parseOptions: s_parseOptions, expectedOutput: @"
+C.M
+Program.Main");
+        }
+
+        [Fact]
+        public void ModuleInitializerMethodIncludedByConditionalAttribute()
+        {
+            string source = @"
+#define INCLUDE
+
+using System;
+using System.Runtime.CompilerServices;
+
+class C
+{
+    [ModuleInitializer, System.Diagnostics.Conditional(""INCLUDE"")]
+    internal static void M() => Console.WriteLine(""C.M"");
+}
+
+class Program 
+{
+    static void Main() => Console.WriteLine(""Program.Main"");
+}
+
+namespace System.Runtime.CompilerServices { class ModuleInitializerAttribute : System.Attribute { } }
+";
+            CompileAndVerify(source, parseOptions: s_parseOptions, expectedOutput: @"
+C.M
+Program.Main");
+        }
+
+        [Fact]
+        public void ModuleInitializerMethodExcludedByConditionalAttribute()
+        {
+            string source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+class C
+{
+    [ModuleInitializer, System.Diagnostics.Conditional(""EXCLUDE"")]
```

Comment:
>[ModuleInitializer, System.Diagnostics.Conditional(""EXCLUDE"")] [](start = 4, length = 64)

Consider also testing different order of attributes.

Reply:
Updated.